src/components/ToolMenu/Draw/Attributions/index.tsx

```diff
@@ -1,67 +1,134 @@
+  fields?: [{
+    name?: string;
+    value?: string;
+  }];
+};
 
 export interface AttributionDrawerProps extends DrawerProps {
   onCustomClose?: (open: boolean) => void;
 }
 
 const AttributionDrawer: React.FC<AttributionDrawerProps> = ({
   onCustomClose,
+  open,
   onClose,
   ...passThroughProps
 }) => {
   const [selectedFeature, setSelectedFeature] = useState<OlFeature>();
   const [isFormValid, setIsFormIsValid] = useState(true);
-  const [currentProperties, setCurrentProperties] = useState<Record<string, any>>({});
-
-  const [form] = Form.useForm();
+  const [availableFeatureCollectionAttributes, setAvailableFeatureCollectionAttributes] = useState<string[]>([]);
+  const [availableFeatureAttributes, setAvailableFeatureAttributes] = useState<string[]>([]);
+  const [api, contextHolder] = notification.useNotification();
+  const [form] = Form.useForm<FormData>();
 
   const map = useMap();
 
   const {
     t
   } = useTranslation();
 
+  const updateAvailableFeatureAttributes = useCallback((fields: FormData) => {
+    const currentKeys = fields.fields
+      ?.map(field => field?.name?.toLowerCase())
+      .filter(key => key !== undefined);
+
+    setAvailableFeatureAttributes(availableFeatureCollectionAttributes.filter(o => !(currentKeys?.includes(o.toLowerCase()))));
+  }, [availableFeatureCollectionAttributes]);
+
   useEffect(() => {
+    if (!map) {
+      return;
+    }
+
     const properties = selectedFeature?.getProperties();
 
-    setCurrentProperties({ ...properties });
+    form.setFieldValue('fields', []);
 
-    const fs: any = {};
     if (properties) {
-      Object.entries(properties).forEach(([key, value]) => {
-        fs[key] = {
-          name: key,
-          value: value
-        };
-      });
+      const filteredPropertyEntries = Object.entries(properties)
+        .filter(([, value]) => !(value instanceof OlGeometry))
+        .map(([key, value]) => [key, value]);
+      const filteredProperties = Object.fromEntries(filteredPropertyEntries);
+
+      const formProperties = Object.entries(filteredProperties)
+        .map(([key, value]) => {
+          return {
+            name: key,
+            value: value
+          };
+        });
+
+      form.setFieldValue('fields', formProperties);
     }
 
-    form.setFieldValue('fields', {});
-    form.setFieldsValue({
-      fields: fs
+    const digitizeLayer = DigitizeUtil.getDigitizeLayer(map);
+    const digitizedFeatures = digitizeLayer.getSource()?.getFeatures();
+
+    const featureCollectionAttributes: Set<string> = new Set();
+    digitizedFeatures?.forEach(feat => {
+      Object.keys(feat.getProperties()).forEach(prop => {
+        if (!(feat.get(prop) instanceof OlGeometry)) {
+          featureCollectionAttributes.add(prop);
+        }
+      });
     });
-  }, [selectedFeature, form]);
+
+    setAvailableFeatureCollectionAttributes(Array.from(featureCollectionAttributes));
+
+    // updateAvailableFeatureAttributes(form.getFieldsValue());
```

Comment:
Line can be removed.

Reply:
done in: [1410fd6](https://github.com/terrestris/shogun-gis-client/pull/1506/commits/1410fd6bc66d3716fe9b90f51323beea1a3978ef)